File: src/main/ngapp/src/app/_models/screen.ts
export class Screen {
  id: number;
  name: string;
  height: number;
  width: number;
  location: object;
  thirdParty: boolean;


  constructor(id: number, name: string, height: number, width: number, location: object, thirdParty: boolean) {
    this.id = id;
    this.name = name;
    this.height = height;
    this.width = width;
    this.location = location;
    this.thirdParty = thirdParty;
  }

  toTable() {
    return {id: this.id, name: this.name, height: this.height, width: this.width, locationId: this.location['id'],
      locationName: this.location['name'], thirdParty: this.thirdParty};
  }
}

File: src/main/java/nl/jtosti/hermes/entities/Screen.java
package nl.jtosti.hermes.entities;

import lombok.*;
import nl.jtosti.hermes.Image;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@ToString
@EqualsAndHashCode
@AllArgsConstructor
public class Screen {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String name;
    private int width;
    private int height;

    @ManyToOne(fetch = FetchType.LAZY)
    private Location location;

    @OneToMany(mappedBy = "screen")
    private List<Image> images = new ArrayList<>();

    @Override
    public String toString() {
        return "<Screen " + Long.toString(id) + ": " + name + ">";
    }
}
